src/Digdir.Domain.Dialogporten.Application/Features/V1/ServiceOwner/Dialogs/Commands/Update/UpdateDialogCommand.cs

```diff
@@ -165,6 +170,13 @@ private async Task AppendActivity(DialogEntity dialog, UpdateDialogDto dto, Canc
     {
         var newDialogActivities = _mapper.Map<List<DialogActivity>>(dto.Activities);
 
+        var organizationLongNames = await _userOrganizationRegistry.GetCurrentUserOrgLongNames(cancellationToken);
+        if (organizationLongNames != null)
```

Comment:
Denne logikken er to steder - begge stedene med en todo om å håndtere dersom vi ikke finner org. Hadde det vært en ide om å refaktorere slik at det er en egen applikasjonstjeneste som har i ansvar å populere aktiviteters PerformedBy (det som gjøres av den statiske metoden `EnsurePerformedByIsSetForActivities`), samt slenge et melding i outbox dersom det ikke lykkes? Denne tjenesten har `IUserOrganizationRegistry` som avhengighet. Tanker?

Reply:
tehe, ja var det vi funderte litt på. Var egentlig satt opp til at alt skulle være abstrahert bort. Før refaktor så kalte vi bare på `EnsurePerformedByIsSetForActivities` som håndterte akkurat det du sier. Men det var definert to steder(UpdateDialogCommand og CreateDialogCommand). Så da var løsningen du skisserte nå et av alternativene. 

Vi landet på å kun abstrahere bort selve opprettelsen av performedBy-entitetene, hovedargumentet var for å slippe å lage en egen applikasjonstjeneste som mest sannsynlig i en stund fremover kun gjør akkurat denne logikken(assumption) + at koden er clean og testbar. Når det er sagt, kan vi nok oppnå samme testbarheten, om ikke bedre om vi skiller ut alt. 

Kan jo være at vi uansett trenger den servicen på sikt til andre ting, så begynner å tenke at det kan være en fornuftig løsning ja.. 

Det blir færre steder å vedlikeholde det, så begynner å helle mot forslaget ditt.. Hva tenker du @oskogstad ? Kanskje jeg har glemt et annet argument her..

Reply:
Refaktorert nå!